tests/Operation/ReplaceOneTest.php

```diff
@@ -31,29 +31,31 @@ public function testConstructorReplacementArgument($replacement): void
         new ReplaceOne($this->getDatabaseName(), $this->getCollectionName(), ['x' => 1], $replacement);
     }
 
-    /** @dataProvider provideUpdateDocuments */
-    public function testConstructorReplacementArgumentRequiresNoOperators($replacement): void
+    /** @dataProvider provideInvalidReplacementDocuments */
+    public function testConstructorReplacementArgumentRequiresDocument($replacement, string $exceptionMessage): void
     {
         $this->expectException(InvalidArgumentException::class);
-        $this->expectExceptionMessage('First key in $replacement argument is an update operator');
+        $this->expectExceptionMessage($exceptionMessage);
         new ReplaceOne($this->getDatabaseName(), $this->getCollectionName(), ['x' => 1], $replacement);
     }
 
     public function provideReplacementDocuments()
     {
         return $this->wrapValuesForDataProvider([
+            [],
             ['y' => 1],
             (object) ['y' => 1],
             new BSONDocument(['y' => 1]),
         ]);
     }
 
-    public function provideUpdateDocuments()
+    public function provideInvalidReplacementDocuments()
```

Comment:
Note: this is going to conflict with https://github.com/mongodb/mongo-php-library/pull/1077. The changes in that PR are more involved, so I'd suggest waiting on that to be merged.

Consideration of PackedArray instances for pipeline values is also going to impact this PR.